pkg/api/types.go

```diff
@@ -1935,6 +1946,28 @@ func (p *Properties) IsNVIDIADevicePluginEnabled() bool {
 	return p.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(common.NVIDIADevicePluginAddonName)
 }
 
+// GetKubernetesVersion returns the cluster Kubernetes version, with the Azure Stack suffix if Azure Stack Cloud.
+func (p *Properties) GetKubernetesVersion() string {
+	if p.IsAzureStackCloud() {
+		return p.OrchestratorProfile.OrchestratorVersion + AzureStackSuffix
+	}
+	return p.OrchestratorProfile.OrchestratorVersion
+}
+
+// GetKubernetesHyperkubeSpec returns the string to use for the Kubernetes hyperkube image.
+func (p *Properties) GetKubernetesHyperkubeSpec() string {
```

Comment:
Looking forward to rebasing on top of this comming and using function in my own refactor PR here:

https://github.com/Azure/aks-engine/pull/2540/files#diff-9d1171fa68cde0d400f802311a34ceccR205

:)